src/test/kotlin/org/jitsi/utils/TemplatedUrlTest.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright @ 2025 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.utils
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.shouldBe
+import java.net.URI
+import java.net.URISyntaxException
+
+class TemplatedUrlTest : ShouldSpec({
+    context("TemplatedUrl") {
+        should("resolve a simple template") {
+            val template = "https://example.com/{{path}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("path", "test")
+
+            templatedUrl.resolve() shouldBe URI("https://example.com/test")
+        }
+
+        should("resolve a template with multiple keys") {
+            val template = "wss://{{host}}/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("host", "example.com")
+            templatedUrl.set("path", "api/resource")
+            templatedUrl.set("param", "value")
+
+            templatedUrl.resolve() shouldBe URI("wss://example.com/api/resource?param=value")
+        }
+
+        should("resolve a template with multiple keys when they are not set()") {
+            val template = "wss://{{host}}/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("host", "example.com")
+            templatedUrl.set("path", "api/resource")
+            templatedUrl.set("param", "value")
+
+            templatedUrl.resolve(
+                mapOf("host" to "example.com", "path" to "api/resource", "param" to "value")
```

Comment:
Do you want these to have different values to ensure that it's picking up the right ones?